code-challenges.test.js

```diff
@@ -45,23 +103,105 @@ const studyMinutesWeek2 = {
+      15, 15, 20, 30, 30, 60, 90,
+    ]);
+    expect(ascendingArray(studyMinutesWeek2)).toEqual([
+      10, 15, 20, 45, 60, 65, 100,
+    ]);
+    expect(ascendingArray(studyMinutesWeek3)).toEqual([
+      -122, -30, 10, 65, 100, 200, 260,
+    ]);
+  });
+});
+
+//ReferenceError: ascendingArray is not defined
+
 // b) Create the function that makes the test pass.
 
+// ============== <<<<< PSEUDO CODE >>>>> ===========================
+//we need a function that will take in an object and pull the desired values (numbers)
+//since this is not an array, we  need to use Object.values function to access the values only
+//we are going to chain .forEach function so we can iterate the obj elements. Each value/element will be pushed into our newArray, one at a time.
+//the function will return sorted version of our new array
+
+const ascendingArray = (obj) => {
+  const newArray = [];
+  Object.values(obj).forEach((val) => {
+    newArray.push(val);
+  });
+  return newArray.sort((a, b) => a - b);
+};
+
+// test passed
+
+// ===================================== CHALLENGE #3 =================================================
+
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const accountTransactions1 = [100, -17, -23, -9]
+// const accountTransactions1 = [100, -17, -23, -9];
 // Expected output: [100, 83, 60, 51]
 
-const accountTransactions2 = [250, -89, 100, -96]
+// const accountTransactions2 = [250, -89, 100, -96];
 // Expected output: [250, 161, 261, 165]
 
-const accountTransactions3 = []
+// const accountTransactions3 = [];
 // Expected output: []
 
 // b) Create the function that makes the test pass.
+
+describe("cumSumArray", () => {
+  it("the function takes in an array and returns an array of the accumulating sum. An empty array should return an empty array", () => {
+    const accountTransactions1 = [100, -17, -23, -9];
+    const accountTransactions2 = [250, -89, 100, -96];
+    const accountTransactions3 = [];
+    const accountTransactions4 = [0];
+    expect(cumSumArray(accountTransactions1)).toEqual([100, 83, 60, 51]);
+    expect(cumSumArray(accountTransactions2)).toEqual([250, 161, 261, 165]);
+    expect(cumSumArray(accountTransactions3)).toEqual([]);
+    expect(cumSumArray(accountTransactions4)).toEqual([0]);
+  });
+});
+
+//ReferenceError: sumSumArray is not defined
+
+// ============== <<<<< PSEUDO CODE >>>>> ===========================
+// the function will receive an array and return the array of same length. We can use .map to iterate
+// we need to work on logic regarding calculations of our sum for each iteration and push into our new array
+// we need to declare two variables and assign "0" to one of them to help with first .map iteration
+
+const cumSumArray = (arr) => {
```

Comment:
Awesome work!!  Really happy to see you using higher-order functions on this problem! 👏